lektor_webpack_support.py

```diff
@@ -24,9 +26,13 @@ def run_webpack(self, watch=False):
         return portable_popen(args, cwd=webpack_root)
 
     def npm_install(self):
-        reporter.report_generic('Running npm install')
-        webpack_root = os.path.join(self.env.root_path, 'webpack')
-        portable_popen(['npm', 'install'], cwd=webpack_root).wait()
+        if self.npm_lock.acquire(False):
+            reporter.report_generic('Running npm install')
+            webpack_root = os.path.join(self.env.root_path, 'webpack')
+            portable_popen(['npm', 'install'], cwd=webpack_root).wait()
+        else:
+            self.npm_lock.acquire()
```

Comment:
To maintain the behavior that this method does not exit until the `npm install` process is complete, since callers expect to have a complete set of installed dependencies after calling it. (I.e., if you can acquire the lock, run npm and then release it. If you can't acquire the lock, wait for it to be released before exiting.) Otherwise you might end up attempting to run `webpack` before it's installed.